src/e2e/puppeteer/__tests__/editable-gap.ts

```diff
@@ -0,0 +1,116 @@
+import clickThought from '../helpers/clickThought'
+import getEditable from '../helpers/getEditable'
+import getEditingText from '../helpers/getEditingText'
+import getSelection from '../helpers/getSelection'
+import hideHUD from '../helpers/hideHUD'
+import paste from '../helpers/paste'
+import { page } from '../setup'
+
+// Intentional overlap between thoughts to fix selection behavior
+// See: useSizeTracking.ts lineHeightOverlap calculation
+// const EXPECTED_OVERLAP = DEFAULT_FONT_SIZE / 8
+
+vi.setConfig({ testTimeout: 20000, hookTimeout: 20000 })
+
+/** Helper function to check if cursor is positioned in the middle of a thought. */
+async function isCursorInMiddle(expectedThought: string): Promise<boolean> {
+  const editingText = await getEditingText()
+  if (editingText !== expectedThought) return false
+
+  const selection = await getSelection()
+  const focusOffset = await selection.focusOffset
+  const textContent = await selection.focusNode?.textContent
+
+  if (!textContent || focusOffset === undefined) return false
+
+  // Check if cursor is not at the very beginning (0) or end (textContent.length)
+  // This guards against padding clicks that set cursor to edges
+  return focusOffset > 0 && focusOffset < textContent.length
+}
+
+/** Helper function to check if selection is lost (dead zone detection). */
+const isSelectionLost = async () => {
+  const selection = await getSelection()
+  const focusNode = await selection.focusNode
+  return !focusNode || focusNode === null
+}
+
+/**
+ * Helper function to test clicking between two thoughts with proper overlap handling.
+ * Validates that clicking in overlapping areas behaves correctly and tests for dead zones.
+ * Tests cursor positioning to ensure clicks don't set cursor to beginning/end of thoughts.
+ */
+async function testClickBetweenThoughts(thought1: string, thought2: string) {
+  const el1 = await getEditable(thought1)
+  const el2 = await getEditable(thought2)
+
+  const rect1 = await el1.boundingBox()
+  const rect2 = await el2.boundingBox()
+
+  if (!rect1 || !rect2) {
+    throw new Error(`Could not get bounding boxes for "${thought1}" and "${thought2}"`)
+  }
+
+  // Calculate overlap (expected to be negative due to intentional overlap)
+  const firstThoughtBottom = rect1.y + rect1.height
+  const secondThoughtTop = rect2.y
+
+  // Note: negative = overlap
+  const overlapHeight = secondThoughtTop - firstThoughtBottom
+
+  // click at the middle of the thought
+  const clickX = rect1.x + rect1.width / 2
+
+  // Test 1: Click in top edge of the overlap zone (should be in first thought)
+  console.info('Test 1: Clicking in top edge of the overlap zone')
```

Comment:
Additional logging in tests is usually unnecessary since a failed assertion logs its line number, telling you exactly how far it executed. Was there another reason for adding this?

Reply:
Okay, thanks!

If you request a review, I would just ask that you let me know which parts are still WIP. Otherwise I will assume it is all ready to go and open to feedback.

Or you can mark it as a draft and I will avoid reviewing until you say it's ready.

Reply:
Understood!